joblib/test/test_numpy_pickle.py

```diff
@@ -279,13 +279,15 @@ def test_masked_array_persistence(tmpdir):
 def test_compress_mmap_mode_warning(tmpdir):
     # Test the warning in case of compress + mmap_mode
     rnd = np.random.RandomState(0)
-    a = rnd.random_sample(10)
+    obj = rnd.random_sample(10)
     this_filename = tmpdir.join("test.pkl").strpath
-    numpy_pickle.dump(a, this_filename, compress=1)
+    numpy_pickle.dump(obj, this_filename, compress=1)
     with warns(UserWarning) as warninfo:
-        numpy_pickle.load(this_filename, mmap_mode="r+")
+        reloaded_obj = numpy_pickle.load(this_filename, mmap_mode="r+")
```

Comment:
There are plenty of tests regarding dump/load that do not check that the returned object is actually correct. This is how this bug slipped through. Those additions to this test would fail on main.